contributors/devel/sig-node/separate-mount-namespace.md

```diff
@@ -0,0 +1,212 @@
+of Kubernetes on a single host without the various mountpoints interfering with
+one another. There may be other inter-instance difficulties besides the mount
+namespace, but this does move in that direction without resorting to more
+complicated solutions like trying to run Kubernetes inside of containers.
+
+## Costs and Implications
+
+If a Kubernetes distribution intends to implement this mount-hiding mode of
+operation, there are a number of costs and implications that should be
+documented for system administrators.
+
+### Privileged pods
+
+Privileged pods that previously expected to have direct access to the host's
+root mount namespace (such as the one spawned by `oc debug`), will now be
+rooted in the new container-runtime mount namespace.  They will still have
+visibility into the host OS's mount points as granted by the normal volumeMount
+permissions, but any mounts made in these privileged pods will not by default
+propagate up to be visible by the Host OS. In order for shells or utilities in
+privileged pods like this to propagate mounts up to the host OS, they can enter
+the host OS namespace by running `nsenter -t 1 -m` (assuming they have the host
+OS's /proc filesystem mounted so they can reach the mount namespace of the real
+init process in /proc/1/ns/mnt)
+
+### Legacy Tools
+
+Any monitoring tool that relies on the ability to run in the host OS and have
+visibility of mountpoints created by Kubelet, the Container Runtime, or
+containers themselves, will need to enter the container mount namespace in
+order to see these mountpoints.  This could be done by:
+
+- Altering the legacy tool to be cloud-native
+- Altering the scripts that launch the legacy tool to run it inside the
+  container-mount namespace
+- Altering the legacy tool to be aware of the container-mount namespace and
+  enter it as needed to probe or interact with the container-specific
+  mountpoints
+- Providing a system-wide mechanism to revert the Kubernetes installation back
+  to the original "all mounts in the Host OS" mode of operation.
+
+### Administrative Access
+
+An administrator logged in to the host OS wishing to obtain information about
+the Kubernetes system for debugging or auditing purposes will need to be aware
+of the new mount namespace, and will need to run 'nsenter' to view or interact
+with these mountpoints.
+
+## Implementation Suggestions
+
+This does not require a change to Kubelet, the Container Runtime, or any other
+existing component, but can be accomplished entirely through changes to the
+environment and init scripts which launch Kublete and the Container Runtime.
+
+### Mount Namespace Creation
+
+The new mount namespace must be a `slave+shared` mountpoint (according to
+[mount_namespace(7)](https://man7.org/linux/man-pages/man7/mount_namespaces.7.html))
+descended from the top-level mount namespace.
+- It needs to be a `slave` to the host OS namespace so it sees the host
```

Comment:
Unfortunately this terminology has not been updated upstream so it's required for accuracy in these documents... not sure what the process is with the Inclusive Naming Initiative (INI) to address this sort of thing.

Reply:
I know.  I suspect there will be a lot of historical baggage to clear out that will take some time to rework.

Reply:
I was actually hoping the reference to the namespace above would be a bit of a "Sorry we couldn't change that on our own" pointer.

Reply:
I asked on the INI mailing list, and I think the most reasonable suggestion I saw there was to refer to the legacy term once and use preferred terms thereafter, except as needed for technical accuracy.  I'll re-read this through and see if I can do better...

Reply:
Yes, I have removed one instance of "slave", and I think it's still clear and accurate.